networkdb/cluster.go

```diff
@@ -44,6 +44,8 @@ func (l *logWriter) Write(p []byte) (int, error) {
 
 // SetKey adds a new key to the key ring
 func (nDB *NetworkDB) SetKey(key []byte) {
+	nDB.Lock()
+	defer nDB.Unlock()
 	logrus.Debugf("Adding key %s", hex.EncodeToString(key)[0:5])
```

Comment:
I intentionally put it inside the lock so that it gets printed when this function effectively does the job